Core/Services/StitchCoreLocalMongoDBService/Sources/StitchCoreLocalMongoDBService/CoreLocalMongoDBService.swift

```diff
@@ -28,33 +29,35 @@ public final class CoreLocalMongoDBService {
 
     public func client(withKey key: String,
                        withDBPath dbPath: String) throws -> MongoClient {
-        let key = "\(Thread.current.hash)_\(key)"
-        if let client = _localInstances[key] {
-            return client
-        }
+        return try cLock.write {
+            let key = "\(Thread.current.hash)_\(key)"
+            if let client = _localInstances[key] {
+                return client
+            }
 
-        objc_sync_enter(self)
-        defer { objc_sync_exit(self) }
+            objc_sync_enter(self)
```

Comment:
[q] If we're using the `cLock`, then is the `objc_sync_enter(self)` no longer necessary?